pages/resources/[page].js

```diff
@@ -1,57 +1,170 @@
+import {
+  getResourcesPromise,
+  getResourcesByCategories,
+  getResourcesByLanguages,
+  getResourcesBySearch,
+} from 'common/constants/api';
+import { Field, Formik } from 'formik';
+import Form from 'components/Form/Form';
+import Input from 'components/Form/Input/Input';
+import omit from 'lodash/omit';
 import styles from '../styles/resources.module.css';
+import ThemedReactSelect from '../../components/Form/Select/ThemedReactSelect';
+import Alert from '../../components/Alert/Alert';
 
 const pageTitle = 'Resources';
 
 function Resources() {
   const router = useRouter();
   const { pathname, query } = router;
-  const currentPage = parseInt(query.page, 10);
-
-  if (!isNumber(currentPage)) {
-    // TODO: Handle situation where user tried some funny business, like `/resources/fuck-you`
-    throw new Error(`'currentPage' is not a number`);
-  }
+  const { page, category, languages, paid, q } = query;
+  const currentPage = parseInt(page, 10);
 
+  const [errorMessage, setErrorMessage] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
+
   const [resources, setResources] = useState([]);
   const [totalPages, setTotalPages] = useState(currentPage);
 
+  const [allCategories, setAllCategories] = useState([]);
+  const [allLanguages, setAllLanguages] = useState([]);
+
+  const [selectedLanguages, setSelectedLanguages] = useState([]);
+
+  const costOptions = [
+    { label: 'Paid', value: true },
+    { label: 'Free', value: false },
+  ];
+
   useEffect(() => {
-    getResourcesPromise({ page: query.page })
+    Promise.all([getResourcesByCategories(), getResourcesByLanguages()])
+      .then(([categoriesResponse, languagesResponse]) => {
+        const {
+          data: { data: categoriesData },
+        } = categoriesResponse;
+        const {
+          data: { data: languagesData },
+        } = languagesResponse;
+
+        setAllLanguages(
+          languagesData.map(languageObject => {
+            return {
+              value: languageObject.name.replace(/ /g, ' ').toLowerCase(),
+              label: languageObject.name,
+            };
+          }),
+        );
+        setAllCategories(
+          categoriesData.map(categoryObject => {
+            return {
+              value: categoryObject.name.replace(/ /g, ' ').toLowerCase(),
+              label: categoryObject.name,
+            };
+          }),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('There was an error finding these resources.');
+      });
+  }, []);
+
+  const handleEndpoint = () => {
+    if (q || paid) {
+      return getResourcesBySearch({ page, q, paid });
+    }
+    if (languages || category) {
+      return getResourcesPromise({ page, category, languages, paid });
+    }
+    return getResourcesBySearch({ page, q, paid });
+  };
+
+  useEffect(() => {
+    setErrorMessage(null);
+
+    // eslint-disable-next-line no-restricted-globals
+    if (isNaN(currentPage)) {
```

Comment:
I think we can just force a redirect to page 1. I doubt a user would care to see this error message and fix out the best thing to do.

Reply:
Ok, yeah makes sense- so you want that to be server side thing in `now.json` using regex to determine if they are requesting an invalid page followed by redirect, or just push to `/resources/1` on client?

Reply:
Just push 👍

Reply:
So this was a bit tricky for me since whether I try to use `isNumber, isNan`, or any other relevant type checker I can think of, even on resources/1 it will flash `NaN` at least once, long enough stick me in an infinite loop. I can't use router outside of `useEffect` to redirect, and the dependency is the query. What do?

Reply:
That makes sense cuz we're not returning, so the rest of the UI attempts to render a tick before the push occurs.

```
if (isNaN(currentPage)) {
  return router.push()
}
```

let's slap this IMMEDIATELY after the `page` is parsed as an integer, so that nothing else evaluates.